crates/workspace/src/workspace.rs

```diff
@@ -4551,22 +4565,38 @@ impl Workspace {
         cx.notify();
     }
 
-    fn serialize_workspace(&mut self, window: &mut Window, cx: &mut Context<Self>) {
+    /// Set the parameter `update` to `true` if the action changes the history of the workspace.
+    /// Typically, there are three cases that change the history:
+    /// 1. A brand new workspace is opened, or an existing workspace has been deleted.
+    /// 2. An existing workspace is loaded, this will change the order of the workspaces in the history.
+    /// 3. `AddFolderToProject` or `RemoveFromProject` action is triggered.
+    fn serialize_workspace(&mut self, update: bool, window: &mut Window, cx: &mut Context<Self>) {
         if self._schedule_serialize.is_none() {
             self._schedule_serialize = Some(cx.spawn_in(window, async move |this, cx| {
                 cx.background_executor()
                     .timer(Duration::from_millis(100))
                     .await;
                 this.update_in(cx, |this, window, cx| {
-                    this.serialize_workspace_internal(window, cx).detach();
+                    this.serialize_workspace_internal(update, window, cx)
+                        .detach();
                     this._schedule_serialize.take();
                 })
                 .log_err();
             }));
         }
     }
 
-    fn serialize_workspace_internal(&self, window: &mut Window, cx: &mut App) -> Task<()> {
+    /// Set the parameter `update` to `true` if the action changes the history of the workspace.
+    /// Typically, there are three cases that change the history:
+    /// 1. A brand new workspace is opened, or an existing workspace has been deleted.
+    /// 2. An existing workspace is loaded, this will change the order of the workspaces in the history.
+    /// 3. `AddFolderToProject` or `RemoveFromProject` action is triggered.
+    fn serialize_workspace_internal(
```

Comment:
I think this 'update' parameter seems pretty sketchy here. Why exactly do we need to record whether this save updates the history?